### PR TITLE
Team statistics - seasonal_summary method

### PR DESCRIPTION
**Pull request by grwthomps**

### CODE REVIEW

File: lib/team_statistics.rb [hunk 1]
```diff
@@ -183,7 +182,6 @@ def biggest_team_blowout(team_id)
       end
     end
     biggest_blowout.to_i
-    # require 'pry'; binding.pry
```
Comment (lrs8810) [inline lib/team_statistics.rb:186]: Thanks for removing these!

File: lib/team_statistics.rb [hunk 2]
```diff
@@ -20,9 +20,8 @@ def best_season(team_id)
       end
     end
 
-    games_by_season = filtered_games.group_by do |game|
-      game.season
-    end
+    # MEMOIZATION
+    games_by_season = filtered_games.group_by(&:season)
```
Comment (hillstew) [inline lib/team_statistics.rb:24]: Neat!

File: lib/team_statistics.rb [hunk 3]
```diff
@@ -240,8 +237,88 @@ def head_to_head(team_id)
...
+    all_seasons = @games.inject([]) do |seasons, (game_id, game)|
+      seasons << game.season
+    end.uniq
+
+
+    team_id = team_id.to_i
+    filtered_games = []
+
+    @games.each do |game_id, game|
+      if game.home_team_id == team_id || game.away_team_id == team_id
```
Comment (lrs8810) [inline lib/team_statistics.rb:252]: Maybe we can use the #inject method to create the new array when we refactor.

File: lib/team_statistics.rb [hunk 4]
```diff
@@ -240,8 +237,88 @@ def head_to_head(team_id)
...
 
+  def seasonal_summary(team_id)
+    all_seasons = @games.inject([]) do |seasons, (game_id, game)|
+      seasons << game.season
+    end.uniq
+
+
+    team_id = team_id.to_i
+    filtered_games = []
+
```
Comment (hillstew) [inline lib/team_statistics.rb:250]: Definitely will be able to refactor filtered games into a helper method!

**Review by hillstew — APPROVED** [review]

File: lib/team_statistics.rb (hunk 2 above)
Comment (tyladevon) [reply lib/team_statistics.rb:24]: We can reuse this refactor in many other areas of our code during refactor time tomorrow.

**Review by lrs8810 — APPROVED** [review]

**Review by tyladevon — APPROVED** [review]

**Pull request merged**